openedge-plugin/src/main/java/org/sonar/plugins/openedge/sensor/OpenEdgeXREFSensor.java

```diff
@@ -149,8 +151,11 @@ public String toString() {
    * Those characters are used in ADM2 applications, and are hard-coded in some procedures.
    */
   public static class InvalidXMLFilterStream extends FilterInputStream {
-    protected InvalidXMLFilterStream(InputStream in) {
+    private final Set<Integer> bytes;
+
+    protected InvalidXMLFilterStream(Set<Integer> skippedBytes, InputStream in) {
       super(in);
+      this.bytes = skippedBytes;
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Store a copy of "skippedBytes". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2384)